lib/cn/cn_tree.c

```diff
@@ -1370,26 +1370,27 @@ cn_node_comp_token_put(struct cn_tree_node *tn)
 static void
 cn_comp_release(struct cn_compaction_work *w)
 {
-    struct kvset_list_entry *le;
-    uint                     kx;
-
     assert(w->cw_node);
 
     if (w->cw_rspill_conc) {
+        struct cn_compaction_work *tmp HSE_MAYBE_UNUSED;
+
         /* This work is on the concurrent spill list.  It should not be
          * released unless it is at head of list.  Verify it is at
          * head and remove it.
          */
-        struct cn_compaction_work *tmp HSE_MAYBE_UNUSED;
-
         mutex_lock(&w->cw_node->tn_rspills_lock);
         tmp = list_first_entry_or_null(&w->cw_node->tn_rspills, typeof(*tmp), cw_rspill_link);
-        assert(tmp == w);
+        if (tmp != w)
+            abort(); /* avoid data integrity corruption */
```

Comment:
Would maybe be nice to expand on how data integrity corruption could occur here.

Reply:
Ok, I'll remove this comment and update the preceding one...

Reply:
Done.